react/utils/navigation.js

```diff
@@ -69,21 +69,14 @@ const normalizeQueryMap = (pathSegments, mapSegments) => {
   }
 }
 
-const normalizeSearchNavigation = (pathSegments, map, segmentsToIgnore) => {
-  const normalizedSegments = pathSegments.map(segment =>
-    segmentsToIgnore.includes(segment) ? segment : segment.toLowerCase()
-  )
-  return { pathSegments: normalizedSegments, map }
-}
-
-const getIgnoredSegments = ignore => {
-  return ignore
-    ? flatten(
-        Object.values(ignore).map(routeModifier =>
-          routeModifier.path.split(PATH_SEPARATOR)
-        )
-      )
-    : []
+const normalizeSearchNavigation = (pathSegments, map, options) => {
+  if (options && options.LOWERCASE === false) {
```

Comment:
Where do this `options` come from? It is a bit weird that there is a `LOWERCASE` prop in it.